### PR TITLE
Add card on file flow and components to quickstart app

### PR DESCRIPTION
**Pull request by dzello**
## Summary

This PR adds new components and flows for storing a card on file for a customer, showing a list of stored cards, and charging a card of their choice. The PR is backwards-compatible in the sense that previous tutorials and flows (charging a card normally) all still work.

To change the app flow from normal card to card on file, the developer must first replace the CUSTOMER_ID in app/Constants.js. A tutorial is coming that explains how to get the CUSTOMER_ID and go through the whole flow.

## Changelog

- Modified HomeScreen to support CardOnFile flow
- Created CardsOnFile* components
- Created ChargeCustomerCard and CreateCustomerCard service wrappers; these call new methods I've added to the quickstart backend

## Test Plan

To test, one should run through the flows on iOS and Android with the CUSTOMER_ID set and without the CUSTOMER_ID set.

### CODE REVIEW

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.js [hunk 1]
```diff
@@ -212,10 +229,54 @@ export default class HomeScreen extends Component {
     }
   }
 
+  async onSelectCardOnFile(cardOnFile) {
+    try {
+      this.showPendingScreen();
+      await chargeCustomerCard(CUSTOMER_ID, cardOnFile.id);
+      showAlert('Your order was successful',
+        'Go to your Square dashbord to see this order reflected in the sales tab.',
```
Comment (okenshields) [inline react-native-in-app-payments-quickstart/app/screens/HomeScreen.js]: nit: misspelt `dashboard`

File: react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+import { CHARGE_CUSTOMER_CARD_SERVER_URL } from '../Constants';
+import ChargeCustomerCardError from '../ChargeCustomerCardError';
+
+export default async function createCustomerCard(customer_id, customer_card_id) {
```
Comment (okenshields) [inline react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.js]: Should the function here be `chargeCustomerCard`?

File: react-native-in-app-payments-quickstart/app/screens/HomeScreen.js (hunk 1 above)
Comment (dzello) [reply react-native-in-app-payments-quickstart/app/screens/HomeScreen.js]: Good catch!

File: react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.js (hunk 2 above)
Comment (dzello) [reply react-native-in-app-payments-quickstart/app/service/ChargeCustomerCard.js]: Yes, will update.

**Comment by dzello** [comment]
@okenshields Pushed commits to resolve those comments.

**Comment by okenshields** [comment]
@dzello , LGTM. Please also resolve the conflicts.
@mootrichard , looks like your commits are also in this PR. So it requires you to sign CLA as well.

**Comment by mootrichard** [comment]
Yeah, the problem is that I've _signed_ the CLA, but I messed up on those two commits and had the wrong email associated with those commits. They're unfortunately under something like `Richard Moot <moot.local>` so it doesn't recognize those as _me_. 😞

**Comment by dzello** [comment]
@okenshields I've resolved the conversations.

**Comment by dzello** [comment]
@okenshields the conflict with the `project.pbxproj` file came from [this commit](https://github.com/square/in-app-payments-react-native-plugin/pull/18/commits/c5c92705dd3a89ba939abf473a601643f186de60). @mootrichard i'm not an expert on this file type, do you know how to merge it?

**Comment by mootrichard** [comment]
I sent you a PR that should hopefully fix my commits and the merge conflict https://github.com/DeveloperMode/in-app-payments-react-native-plugin/pull/3

**Comment by dzello** [comment]
Merged in the PR. Looks like we got an error on Travis:

```
error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
```

Then we just need to figure out the CLA for @mootrichard too.

**Comment by mootrichard** [comment]
@dzello – I'm not sure what I did here now. 😞 It looks like when I created the https://github.com/DeveloperMode/in-app-payments-react-native-plugin/pull/3 I might have caused several files to get removed. Just looking at the files changed I can see that it looks totally incorrect.

**Comment by dzello** [comment]
@mootrichard No worries! I went ahead and reverted those changes but kept the ones you made around the style guide. Then I merged in master from upstream. Upstream had a more recent version of `react-native-in-app-payments-quickstart/ios/RNInAppPaymentsQuickstart.xcodep
roj/project.pbxproj` so I just took that version in, since I don't believe it was something we'd intentionally meant to change.

**Comment by dzello** [comment]
I pushed one more commit to update the `react-native-in-app-payments-quickstart/ios/RNInAppPaymentsQuickstart.xcodep roj/project.pbxproj` file to a version that works with `react-native run-ios` command. With the changes from master it was still failing with the `Podfile.lock` error from above. As of this commit it is working and I can load the quickstart app in the iOS simulator.

**Comment by mootrichard** [comment]
@okenshields So it looks like Travis is failing the build due to the logs exceeding the limit. Is there something that we could do to mitigate how much it is polluting the logs?

**Comment by okenshields** [comment]
> @okenshields So it looks like Travis is failing the build due to the logs exceeding the limit. Is there something that we could do to mitigate how much it is polluting the logs?

Looks like the latest commit caused significant increase in logs. Taking a look..

**Comment by okenshields** [comment]
@dzello , there seems to be some issue with your latest commit (https://github.com/square/in-app-payments-react-native-plugin/pull/18/commits/baac5bbbf0a2c3d59459abb377b39e030941fb15). Those libraries shouldn't be needed in the project file anymore after RN 0.60.0. Please see https://facebook.github.io/react-native/blog/2019/07/03/version-60#native-modules-are-now-autolinked or RN upgrade guide: https://react-native-community.github.io/upgrade-helper/?from=0.57.8&to=0.60.5 for more details, in particular https://github.com/react-native-community/upgrade-helper/issues/47. Also I tried latest master and had no issue running ```react-native run-ios```.

**Comment by okenshields** [comment]
> @dzello , there seems to be some issue with your latest commit ([baac5bb](https://github.com/square/in-app-payments-react-native-plugin/commit/baac5bbbf0a2c3d59459abb377b39e030941fb15)). Those libraries shouldn't be needed in the project file anymore after RN 0.60.0. Please see https://facebook.github.io/react-native/blog/2019/07/03/version-60#native-modules-are-now-autolinked or RN upgrade guide: https://react-native-community.github.io/upgrade-helper/?from=0.57.8&to=0.60.5 for more details, in particular [react-native-community/upgrade-helper#47](https://github.com/react-native-community/upgrade-helper/issues/47). Also I tried latest master and had no issue running `react-native run-ios`.

I just tried out this branch locally. I didn't test all the new functionalities but I was able to build and `react-native run-ios` successfully after reverting the changes made to `project.pbxproj`.

**Comment by dzello** [comment]
@okenshields @mootrichard I've gone ahead and reverted the `project.pbxproj` change and also merged in the latest commits from master.

**Comment by mootrichard** [comment]
@dzello – Thank you for getting these updated!
@okenshields – It looks like we are passing CI/CD now

Is there anything else that needs to be reviewed for getting this merged?

**Comment by okenshields** [comment]
LGTM. Thanks a lot!

**Pull request merged**